pyfar/signals/deterministic.py

```diff
@@ -256,6 +393,491 @@ def exponential_sweep_time(n_samples, frequency_range, n_fade_out=90,
     return signal
 
 
+def exponential_sweep_freq(
+        n_samples, frequency_range, start_margin, stop_margin, fade_in=None,
+        fade_out=None, butterworth_order=8, sampling_rate=44100,
+        group_delay=False):
+    """
+    Generate sine sweep with exponentially increasing frequency in the
+    frequency domain.
+
+    Sweep synthesis according to [#]_.
+
+    The exponential sweep can also be generated in the time domain
+    (:py:func:`~exponential_sweep_time`). Frequency domain synthesis
+    exhibits smooth magnitude spectra in trade of a slightly irregular
+    temporal envelope. Time domain synthesis exhibits a constant temporal
+    envelope in trade of slight ripples in the magnitude response.
+
+    .. note::
+        The envelope of the sweep time signal should be constant, apart from
+        slight overshoots at the beginning and end. If this is not the case,
+        try to increase `n_samples`, `start_margin`, `stop_margin`, `fade_in`
+        or `fade_out`.
+
+    Parameters
+    ----------
+    n_samples : int
+        The length of the sweep in samples.
+    frequency_range : array like
+        Frequency range of the sweep given by the lower and upper cut-off
+        frequency in Hz. The restriction of the frequency range is realized
+        by applying a Butterworth high-pass if ``frequency_range[0] > 0``
+        and/or by a low-pass if ``frequency_range[1] < sampling_rate / 2``.
+        Note that the exponential sweep can not start at 0 Hz, because its
+        magnitude is defined by 1/frequency.
+    start_margin : int, float
+        The time in samples, at which the sweep starts. The start margin is
+        required because the frequency domain sweep synthesis has pre-ringing
+        in the time domain.
+    stop_margin : int, float
+        Time in samples, at which the sweep stops. This is relative to
+        `n_samples`, e.g., a stop margin of 100 samples means that the sweep
+        ends at sample ``n_samples-100``. This is required, because the
+        frequency domain sweep synthesis has post-ringing in the time domain.
+    fade_in : int, optional
+        Duration of a squared sine fade-in in samples. The fade starts at the
+        first sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``None`` does not apply
+        a fade-in.
+    fade_out : int, optional
+        Duration of a squared cosine fade-out in samples. The fade ends at the
+        last sample of the sweep that is closer than 60 dB to the absolute
+        maximum of the sweep time signal. The default ``None`` does not apply
+        a fade-out.
+    butterworth_order : int, optional
+        The order of the Butterworth filters that are applied to limit the
+        frequency range (see above). The default is ``8``.
```

Comment:
I'd suggest renaming this to bandpass order or similar to make this more intuitive.

Reply:
done